images/cert-manager-webhook-pdns/tests/main.tf

```diff
@@ -0,0 +1,78 @@
+terraform {
+  required_providers {
+    oci       = { source = "chainguard-dev/oci" }
+    imagetest = { source = "chainguard-dev/imagetest" }
+  }
+}
+
+variable "digest" {
+  description = "The image digest to run tests over."
+}
+
+data "oci_string" "ref" { input = var.digest }
+
+data "oci_exec_test" "help" {
+  digest = var.digest
+  script = "docker run --rm $IMAGE_NAME -h"
+}
+
+data "imagetest_inventory" "this" {}
+
+resource "imagetest_harness_k3s" "this" {
+  name      = "cert-manager-webhook-pdns"
+  inventory = data.imagetest_inventory.this
+}
+
+resource "random_pet" "suffix" {}
+
+module "helm" {
+  source = "../../../tflib/imagetest/helm"
+
+  name      = "cert-manager-webhook-pdns-${random_pet.suffix.id}"
+  namespace = "cert-manager-webhook-pdns-${random_pet.suffix.id}"
+  repo      = "https://zachomedia.github.io/cert-manager-webhook-pdns"
+  chart     = "cert-manager-webhook-pdns"
+  timeout   = "600s"
+
+  values = {
+    image = {
+      repository = data.oci_string.ref.registry_repo
+      tag        = data.oci_string.ref.pseudo_tag
+    }
+  }
+}
+
+module "helm-cert-manager" {
+  source = "../../../tflib/imagetest/helm"
+
+  name    = "cert-manager"
+  repo    = "https://charts.jetstack.io"
+  chart   = "cert-manager"
+  timeout = "600s"
+
+  values = {
+    installCRDs = "true"
+  }
+}
+
+resource "imagetest_feature" "basic" {
+  harness     = imagetest_harness_k3s.this
+  name        = "basic"
+  description = "Basic functionality of cert-manager-webhook-pdns installed in a Kubernetes cluster"
+
+  steps = [
+    {
+      # App needs cert-manager and its CRDs as a prerequisite.
+      name = "cert-manager install"
+      cmd  = module.helm-cert-manager.install_cmd
+    },
+    {
+      name = "Helm install"
+      cmd  = module.helm.install_cmd
+    }
+  ]
+
+  labels = {
+    type = "k8s",
+  }
+}
```

Comment:
Can we add a shell test to validate functionality? i.e after installing into a cluster, add [powerDNS helm chart](https://artifacthub.io/packages/helm/halkeye/powerdns) and a test for this image?

Or if theres a technical blocker from doing that, a TESTING.md file with manual validation steps?

Reply:
Actually I did trust their Readiness probes but I'll dig into this see what can we do here. That Helm Chart seems a bit oldie but will check. No blockers. 👍

Reply:
Yea good point, or maybe just spin up the container if the chart is old. But if you think it's not feasible to fully automate some manual steps for validation would be awesome :)